src/jupyterhub_config/custom_spawner.py

```diff
@@ -179,6 +181,21 @@ def _configure_environment(self, user_dir: Path, user_env_dir: Path, username: s
         self.environment['PYTHONSTARTUP'] = os.path.join(os.environ['JUPYTERHUB_CONFIG_DIR'], 'startup.py')
         self.environment['JUPYTERHUB_USER'] = username
 
+        group_names = [group.name for group in self.user.groups]
+        self.log.info(f'User {self.user.name} groups: {group_names}')
+
+        if self.user.admin or self.RW_MINIO_GROUP in group_names:
+            self.log.info(f'MinIO read/write user detected: {username}. Setting up minio_rw credentials.')
+            self.environment['MINIO_ACCESS_KEY'] = self.environment['MINIO_RW_ACCESS_KEY']
+            self.environment['MINIO_SECRET_KEY'] = self.environment['MINIO_RW_SECRET_KEY']
+        else:
+            self.log.info(f'Non-admin user detected: {username}. Removing admin credentials.')
+            self.environment.pop('MINIO_RW_ACCESS_KEY', None)
+            self.environment.pop('MINIO_RW_SECRET_KEY', None)
+
+        # TODO: add a white list of environment variables to pass to the user's environment
+        self.environment.pop('JUPYTERHUB_ADMIN_PASSWORD', None)
```

Comment:
not yet. I was thinking to leave it in Rancher so that people with Rancher access can log in with admin user. Also I haven't figured it out but I want to build this spark_user automatically.